bcda/cli_test.go

```diff
@@ -176,4 +169,102 @@ func (s *CLITestSuite) TestGetCCLFFileMetadata() {
+	assert.NotNil(err)
+
+	metadata, err = getCCLFFileMetadata("/cclf/T.ABCDE.ACO.ZC8Y18.D181120.T1000010")
+	assert.NotNil(err)
+
+}
+func (s *CLITestSuite) TestSortCCLFFiles() {
+	assert := assert.New(s.T())
+	var cclf0, cclf8, cclf9 []cclfFileMetadata
+
+	filePath := "../shared_files/cclf/"
+	err := filepath.Walk(filePath, sortCCLFFiles(&cclf0, &cclf8, &cclf9))
+	assert.Nil(err)
+	assert.Equal(1, len(cclf8))
+	assert.Equal(1, len(cclf9))
+	assert.Equal(0, len(cclf0))
+
+	cclf0, cclf8, cclf9 = nil, nil, nil
+
+	filePath = "../shared_files/cclf_BadFileNames/"
+	err = filepath.Walk(filePath, sortCCLFFiles(&cclf0, &cclf8, &cclf9))
+	assert.Nil(err)
+	assert.Equal(1, len(cclf8))
+	assert.Equal(0, len(cclf9))
+	assert.Equal(0, len(cclf0))
+
+	cclf0, cclf8, cclf9 = nil, nil, nil
+
+	filePath = "../shared_files/cclf0/"
+	err = filepath.Walk(filePath, sortCCLFFiles(&cclf0, &cclf8, &cclf9))
+	assert.Nil(err)
+	assert.Equal(0, len(cclf8))
+	assert.Equal(0, len(cclf9))
+	assert.Equal(3, len(cclf0))
+
+	cclf0, cclf8, cclf9 = nil, nil, nil
+
+	filePath = "../shared_files/cclf8/"
+	err = filepath.Walk(filePath, sortCCLFFiles(&cclf0, &cclf8, &cclf9))
+	assert.Nil(err)
+	assert.Equal(5, len(cclf8))
+	assert.Equal(0, len(cclf9))
+	assert.Equal(0, len(cclf0))
+
+	cclf0, cclf8, cclf9 = nil, nil, nil
+
+	filePath = "../shared_files/cclf9/"
+	err = filepath.Walk(filePath, sortCCLFFiles(&cclf0, &cclf8, &cclf9))
+	assert.Nil(err)
+	assert.Equal(0, len(cclf8))
+	assert.Equal(4, len(cclf9))
+	assert.Equal(0, len(cclf0))
+
+	cclf0, cclf8, cclf9 = nil, nil, nil
+
+	filePath = "../shared_files/cclf_All/"
+	err = filepath.Walk(filePath, sortCCLFFiles(&cclf0, &cclf8, &cclf9))
+	assert.Nil(err)
+	assert.Equal(5, len(cclf8))
+	assert.Equal(4, len(cclf9))
+	assert.Equal(3, len(cclf0))
+	for _, metadata := range cclf8 {
+		assert.Equal(8, metadata.cclfNum)
+	}
+	for _, metadata := range cclf9 {
+		assert.Equal(9, metadata.cclfNum)
+	}
+	for _, metadata := range cclf0 {
+		assert.Equal(0, metadata.cclfNum)
+	}
+}
+
+func (s *CLITestSuite) TestImportCCLFDirectory() {
+	assert := assert.New(s.T())
+
+	// set up the test app writer (to redirect CLI responses from stdout to a byte buffer)
+	buf := new(bytes.Buffer)
+	s.testApp.Writer = buf
+
+	args := []string{"bcda", "import-cclf-directory", "--directory", "../shared_files/cclf/"}
+	err := s.testApp.Run(args)
+	assert.Nil(err)
+	assert.Contains(buf.String(), "Completed CCLF import.")
+	assert.Contains(buf.String(), "Successfully imported 2 files.")
+	assert.Contains(buf.String(), "Failed to import 0 files.")
+	buf.Reset()
+
+	// dir has 3 files, but 2 will be ignored because of bad file names.
+	args = []string{"bcda", "import-cclf-directory", "--directory", "../shared_files/cclf_BadFileNames/"}
+	err = s.testApp.Run(args)
+	assert.Nil(err)
+	assert.Contains(buf.String(), "Completed CCLF import.")
+	assert.Contains(buf.String(), "Successfully imported 1 files.")
+	assert.Contains(buf.String(), "Failed to import 0 files.")
```

Comment:
Wondering if we need to be informed that we failed to import 2 files with bad filenames.  Looking at the end results, it looks like we had no problems (i.e., no failures to import files).  But in reality, we actually had issues with 2 files.

Reply:
I added in a skipped count to the return